### PR TITLE
[3.4] Feature/rebootid notice changes, backport of #9523

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Backport of #9523, already includes https://github.com/arangodb/arangodb/pull/9686.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5657/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5667/

**Comment by KVS85** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5885/

File: arangod/Cluster/ServerState.cpp [hunk 1]
```diff
@@ -329,6 +333,32 @@ bool ServerState::unregister() {
...
+{
+  TRI_ASSERT(!_id.empty());
+  std::string rebootIdPath = "Current/ServersKnown/" + _id + "/rebootId";
+  AgencyCommResult result = comm.getValues(rebootIdPath);
+
+  if (!result.successful()) {
+    LOG_TOPIC(WARN, Logger::CLUSTER)
+      << "Could not read back " << rebootIdPath;
+
+    ResultT<uint64_t>::error(TRI_ERROR_INTERNAL, "could not read rebootId from agency");
```
Comment (markuspf) [inline arangod/Cluster/ServerState.cpp]:
```suggestion
    return ResultT<uint64_t>::error(TRI_ERROR_INTERNAL, "could not read rebootId from agency");
```

See also the fixes for `devel` and `3.5`

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6069/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6141/
separate windows run:
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-windows/248/
The one test failure on windows is due to a faulty test, being worked on in https://github.com/arangodb/arangodb/pull/9958.

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6188/

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6346/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6349/

**Comment by goedderz** [comment]
Jenkins is green.

File: tests/V8Server/v8-users-test.cpp [hunk 2]
```diff
@@ -25,6 +25,16 @@
 #include "src/objects-inl.h" // (required to avoid compile warnings) must inclide V8 _before_ "catch.cpp' or CATCH() macro will be broken
 #include "src/objects/scope-info.h" // must inclide V8 _before_ "catch.cpp' or CATCH() macro will be broken
 
+#ifdef _WIN32
+// V8 breaks several constants used by windows headers. This later introduces compile errors in other includes,
+// e.g. via
+// ShardingFeature.h -> ShardingStrategy.h -> ClusterInfo.h -> RebootTracker.h -> Scheduler.h -> Basics/asio_ns.h -> <asio/error.hpp>
+// which, at some point, includes mswsockdef.h and mswsock.h.
+// This fixes these errors.
+#define CONST const
```
Comment (mchacki) [inline tests/V8Server/v8-users-test.cpp]: :cry:

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+include_directories(.)
```
Comment (mchacki) [inline tests/CMakeLists.txt]: Is this intentional?

**Comment by mchacki** [comment]
Confirm: Jenkins is good

File: tests/CMakeLists.txt (hunk 3 above)
Comment (goedderz) [reply tests/CMakeLists.txt]: Yes. It's backported from devel and allows for includes relative to `tests/`, e.g. `#include "Mocks/Servers.h"`, even if you aren't in the `Mocks` directory.

**Pull request merged**